sql/core/src/test/scala/org/apache/spark/sql/execution/UnsafeRowSerializerSuite.scala

```diff
@@ -40,9 +44,15 @@ class ClosableByteArrayInputStream(buf: Array[Byte]) extends ByteArrayInputStrea
 class UnsafeRowSerializerSuite extends SparkFunSuite {
 
   private def toUnsafeRow(row: Row, schema: Array[DataType]): UnsafeRow = {
-    val internalRow = CatalystTypeConverters.convertToCatalyst(row).asInstanceOf[InternalRow]
+    val converter = unsafeRowConverter(schema)
+    converter(row)
+  }
+
+  private def unsafeRowConverter(schema: Array[DataType]): Row => UnsafeRow = {
```

Comment:
Yes, I got your mean, if we inline that in `toUnsafeRow`, then for every call of `toUnsafeRow`, we will get a new instance of `Converter` according to the schema, this is actually very expensive, as it's codegen internally for creating the `converter` instance.

Probably we'd better to remove the function `toUnsafeRow` in the future, since it's always cause performance problem, and people even not notice that.